src/loan.rs

```diff
@@ -827,329 +896,81 @@ impl Lender1 {
 
         let mut liquidation_transaction = Transaction {
             version: 2,
-            lock_time: 0,
+            lock_time: self.timelock as u32,
             input: tx_ins,
             output: tx_outs,
         };
 
         // fulfill collateral input covenant script to liquidate the position
-        let sighash = SigHashCache::new(&liquidation_transaction).segwitv0_sighash(
-            0,
-            &self.collateral_script.clone(),
-            self.repayment_collateral_input.original_txout.value,
-            SigHashType::All,
-        );
-
-        let sig = SECP256K1.sign(
-            &elements::secp256k1_zkp::Message::from(sighash),
-            &self.keypair.0,
-        );
-
-        let mut sig = sig.serialize_der().to_vec();
-        sig.push(SigHashType::All as u8);
-        let if_flag = vec![];
-
-        liquidation_transaction.input[0].witness = TxInWitness {
-            amount_rangeproof: None,
-            inflation_keys_rangeproof: None,
-            script_witness: vec![sig, if_flag, self.collateral_script.to_bytes()],
-            pegin_witness: vec![],
-        };
-
-        Ok(liquidation_transaction)
-    }
-}
-
-fn loan_contract(
```

Comment:
You can probably also remove https://github.com/comit-network/baru/blob/master/src/loan/stack_simulator.rs now :)